File: BucketList/app/src/main/java/com/yarin/bucketlist/MainActivity.java
package com.yarin.bucketlist;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupClickListeners();
    }



    private void setupClickListeners() {
        final CardView thingsToDoCard = findViewById(R.id.card_thingstodo);
        final CardView placesToGoCard = findViewById(R.id.card_placestogo);

        thingsToDoCard.setOnClickListener(v -> startActivity(new Intent(this, ThingsToDoActivity.class)));
        placesToGoCard.setOnClickListener(v -> startActivity(new Intent(this, PlacesToGoActivity.class)));

    }
}

File: MarketingApp/app/src/main/java/com/yarin/marketingapp/MainActivity.kt
package com.yarin.marketingapp

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import com.yarin.marketingapp.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.buttonPreview.setOnClickListener {
            onPreviewClicked()
        }

        val spinnerValues: Array<String> = arrayOf("Android Developer", "Android Engineer")
        val spinnerAdapter =
            ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, spinnerValues)
        binding.spinner.adapter = spinnerAdapter
    }

    private fun onPreviewClicked() {

        val message = Message(
            binding.editTextContactName.text.toString(),
            binding.editTextContactNumber.text.toString(),
            binding.editTextContactDisplayname.text.toString(),
            binding.checkBoxJunior.isChecked,
            binding.spinner.selectedItem?.toString(),
            binding.checkBoxImmediateStart.isChecked,
            binding.editTextStartDate.text.toString()
        )

        val previewActivityIntent = Intent(this, PreviewActivity::class.java)
        previewActivityIntent.putExtra("message", message)
        startActivity(previewActivityIntent)
    }


}